mpi-proxy-split/doc/nimrod-build-tutorial.txt

```diff
@@ -0,0 +1,273 @@
+NIMROD's build process. If the packages need to be rebuilt (for instance, a
+debug version of MPICH is desired), they can be built separately.
+
+It also contains logs on the previous build processes. To debug the build
+process, it's helpful to look at the latest mknimall.log, which will show the
+following:
+
+  SUMMARY
+
+  illios executed ./mknimall.sh -t -j 8 on reached at Fri Apr 22 11:26:44 PDT 2022.
+
+  SUCCESS - Installations: ...
+
+  Configure successes: ...
+  Configure failures: ... 
+  Build successes: ...
+  Build failures: ... 
+  Installation successes: ... 
+  Installation failures: ...
+  No tests were run.
+
+Here, we can gather which packages failed, and where in the process. The log
+also gives more detailed information in each sub-section:
+
+    Bilder examining netcdf_fortran (2022-04-06-23:47:45)
+  ========================================================
+  buildNetcdf_fortran already known.
+  [buildChain] Dependencies are NETCDF_FORTRAN_DEPS = netcdf hdf5 cmake.
+  [buildChain] Builds are NETCDF_FORTRAN_BUILDS = sersh par.
+
+  [buildChain] EXECUTING buildNetcdf_fortran
+  ...
+  Package netcdf_fortran-4.2-par failed to configure at 2022-04-06-23:47:51.
+
+  [buildChain] EXECUTING testNetcdf_fortran
+  ...
+
+  [buildChain] EXECUTING installNetcdf_fortran
+  ...
+
+  [buildChain] EXECUTING findNetcdf_fortran
+  ...
+
+Bilder checks the installation of each package in three steps and reports its
+status at the end of each step. In the example above, netcdf_fortran failed to
+configure. nimall/BUILD summarizes the debugging steps succinctly:
+
+  For a successfully installed package there are three shell scripts created in
+  the build directory (see below) corresponding to each step. These scripts are
+  named *-config.sh, *-build.sh and *-install.sh where the wildcard value is
+  machine and package dependent. There are also three output files corresponding
+  to each step, *-config.txt, *-build.txt and *-install.txt. When a package
+  fails on a step, the first thing to check is the error located in the output
+  file corresponding to the failed step.
+
+Some issues that you may face are:
+  * conflicting gfortran packages ..... devtoolset-8-gcc-gfortran should be
+                                        necessary.
+  * conflicting BLAS packages ......... blas-static and openblas-static should
+                                        be necessary. Note that the former
+                                        doesn't exist on CentOS Stream.
+  * PyTables not installed ............ Install PyTables for Python 3:
+                                          python3 -m pip install --user packaging
+                                          python3 -m pip install --user numpy
+                                          python3 -m pip install --user tables
+  * superlu_dist7 built but not found . See the -F flag above more details.
+
+## SUBSEQUENT BUILDS ##
```

Comment:
`SUBSEQUENT BUILDS` -> `APPENDIX FOR DEVELOPERS:  MODIFYING AND RE-BUILDING NIMROD `